wolvic/browser/vr/wvr_manager.cc

```diff
@@ -481,11 +508,22 @@ void WvrManager::SetInputSourceButtonListener(
 bool WvrManager::SubmitFrameInternal(int16_t frame_index) {
   DCHECK(IsOnWvrThread());
 
+  // Force exit WebXR before pushing the frame if the presenting generation is
+  // changed by stopping presenting in Wolvic.
+  if (presenting_generation_ != system_state_.displayState.presentingGeneration) {
+    if (!get_frame_data_callback_.is_null())
+      std::move(get_frame_data_callback_).Run(nullptr);
+
+    if (exit_vr_callback_)
+      std::move(exit_vr_callback_).Run();
+    return false;
+  }
+
   auto& layer = browser_state_.layerState[0].layer_stereo_immersive;
   layer.frameId = frame_index;
   layer.textureSize.width = surface_size_.width();
   layer.textureSize.height = surface_size_.height();
-  layer.textureHandle = texture_id_;
+  layer.textureHandle = texture_handle_id_;
```

Comment:
seems like a different bug? Personally I have a hard time differentiating `texture_id_` from `texture_handle_id_` without checking the code, it's quite easy to confuse them. Perhaps we need a followup renaming those two.

Reply:
It's the same bug.  `texture_id_`  in the old code was used with gl texture's id and `SurfaceTexture`'s handle. But the new code changes to set the generated ID to `SurfaceTexture` handle to prevent interference with the state of the new webXR from the previous webXR state.  If [this bug](https://github.com/Igalia/wolvic/issues/662) is fixed, it's okay to mix with both of them.